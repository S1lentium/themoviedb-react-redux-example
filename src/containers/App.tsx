import React from "react";
import { connect } from "react-redux";
import { BrowserRouter, Route, Switch } from "react-router-dom";
import { getGenres } from "../actions";
import Layout from "../components/Layout";
import MovieDetails from "./MovieDetails";
import PopularMovies from "./PopularMovies";

interface IDispatchProps {
  getGenres: () => void;
}

export class App extends React.Component<IDispatchProps> {
  public componentDidMount() {
    this.props.getGenres();
  }

  public render() {
    return (
      <BrowserRouter>
        <Layout>
          <Switch>
            <Route exact={true} path="/" component={PopularMovies} />
            <Route path="/movie/:id" component={MovieDetails} />
          </Switch>
        </Layout>
      </BrowserRouter>
    );
  }
}

export default connect(
  null,
  { getGenres },
)(App);
