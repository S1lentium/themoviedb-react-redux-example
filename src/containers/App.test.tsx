/* tslint:disable:no-implicit-dependencies */
import { shallow } from "enzyme";
import React from "react";
import { App } from "./App";

describe("App", () => {
  it("should render", () => {
    const getGenres = () => ({});
    const component = shallow(<App getGenres={getGenres} />);

    expect(component).toMatchSnapshot();
  });
});
