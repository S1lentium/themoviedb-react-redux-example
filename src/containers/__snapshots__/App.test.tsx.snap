// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render 1`] = `
<BrowserRouter>
  <WithStyles(Layout)>
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        path="/movie/:id"
      />
    </Switch>
  </WithStyles(Layout)>
</BrowserRouter>
`;
